### PR TITLE
Update Bitbucket pipeline

### PR DESCRIPTION
**Pull request by ekufta0530**
Updated to match functionality of other pipelines. This is also similar to what we have in-app for the bitbucket CI config. We will be updating this as well.

# Thanks for improving Semgrep Docs 😀

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [ ] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [ ] Redirects are added if the PR changes page URLs
- [ ] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link

### CODE REVIEW

File: src/components/code_snippets/_bitbucket-semgrep-app-sast.mdx [hunk 1]
```diff
@@ -2,31 +2,31 @@
...
+  pull-requests:
+    '**': # This applies to pull requests for all branches
       - step:
-          name: 'Run Semgrep scan with current branch'
-          deployment: dev # https://support.atlassian.com/bitbucket-cloud/docs/set-up-and-monitor-deployments/
           image: semgrep/semgrep
+          name: Semgrep Scan on PR
           script:
-            # The following variables are required to set up a scan connected to Semgrep Cloud Platform:
-            - export SEMGREP_APP_TOKEN=$SEMGREP_APP_TOKEN
```
Comment (armchairlinguist) [inline src/components/code_snippets/_bitbucket-semgrep-app-sast.mdx:13]: I don't think we ought to remove this line :) Can you restore it?

File: src/components/code_snippets/_bitbucket-semgrep-app-sast.mdx [hunk 2]
```diff
@@ -2,31 +2,31 @@
...
-            # Uncomment the following lines if Semgrep Cloud Platform > Findings Page does not create links
-            # to the code that generated a finding or if you are not receiving PR or MR comments.
-            # - export SEMGREP_JOB_URL="${SEMGREP_REPO_URL}/addon/pipelines/home#!/results/${BITBUCKET_PIPELINE_UUID}"
-            # - export SEMGREP_COMMIT=$BITBUCKET_COMMIT
-            # - export SEMGREP_PR_ID=$BITBUCKET_PR_ID
-            # - export SEMGREP_BRANCH=$BITBUCKET_BRANCH
-            # - export SEMGREP_REPO_URL=$BITBUCKET_GIT_HTTP_ORIGIN
-            # - export SEMGREP_REPO_NAME=$BITBUCKET_REPO_FULL_NAME
+            - export BITBUCKET_TOKEN=$PAT # Necessary for PR comments
+            - export SEMGREP_BASELINE_REF="origin/main"
```
Comment (armchairlinguist) [inline src/components/code_snippets/_bitbucket-semgrep-app-sast.mdx:20]:
```suggestion
            # Change to your default branch if different from main
            - export SEMGREP_BASELINE_REF="origin/main"
```

File: src/components/code_snippets/_bitbucket-semgrep-app-sast.mdx [hunk 3]
```diff
@@ -2,31 +2,31 @@
...
+            - git fetch origin "+refs/heads/*:refs/remotes/origin/*"
+            - semgrep ci
 
+  custom:
+  # additinal setup for cron in BitBucket https://support.atlassian.com/bitbucket-cloud/docs/pipeline-triggers/#On-schedule
```
Comment (armchairlinguist) [inline src/components/code_snippets/_bitbucket-semgrep-app-sast.mdx:25]:
```suggestion
  # Trigger job manually. For cron in Bitbucket, see: https://support.atlassian.com/bitbucket-cloud/docs/pipeline-triggers/#On-schedule
```

Comment (armchairlinguist) [reply src/components/code_snippets/_bitbucket-semgrep-app-sast.mdx:25]: Based on our convo I understood this automatically gives a manual trigger, but if that's not right we can adjust the comment accordingly.

File: src/components/code_snippets/_bitbucket-semgrep-app-sast.mdx [hunk 4]
```diff
@@ -2,31 +2,31 @@
 image: atlassian/default-image:latest
 
 pipelines:
-  default:
-    - parallel:
+  branches:
+    main:
```
Comment (armchairlinguist) [inline src/components/code_snippets/_bitbucket-semgrep-app-sast.mdx:6]:
```suggestion
    # Change to your default branch if different from main
    main:
```

File: docs/semgrep-ci/sample-ci-configs.md [hunk 5]
```diff
@@ -254,7 +254,7 @@ To add a Semgrep configuration snippet into Bitbucket Pipelines:
 2. Copy the relevant code snippet provided in [Sample Bitbucket Pipelines configuration snippet](#sample-bitbucket-pipelines-configuration-snippet), and then paste it to your `bitbucket-pipelines.yml`.
 3. Commit the updated `bitbucket-pipelines.yml` configuration file.
 4. The Semgrep job starts automatically upon detecting the committed `bitbucket-pipelines.yml` file. You can also view the job through Bitbucket's interface, by clicking **your repository > Pipelines**.
```
Comment (armchairlinguist) [inline docs/semgrep-ci/sample-ci-configs.md:256]:
```suggestion
4. The Semgrep job starts automatically upon detecting the committed `bitbucket-pipelines.yml` file. You can view the job through Bitbucket's interface, by clicking **your repository > Pipelines**.
```
You didn't change this line, but the "also" doesn't make sense, and this seems like a good time to remove it.

File: docs/semgrep-ci/sample-ci-configs.md [hunk 6]
```diff
@@ -254,7 +254,7 @@ To add a Semgrep configuration snippet into Bitbucket Pipelines:
...
-5. Optional: Create a separate CI job for diff-aware scanning, which scans only changed files in PRs or MRs, by repeating steps 1-3 and uncommenting the `SEMGREP_BASELINE_REF` definition provided within the code snippet.
+5. Create a daily scheduled run for the custom pipeline on the main branch by [scheduling a pipleline in BitBucket](https://support.atlassian.com/bitbucket-cloud/docs/pipeline-triggers/#On-schedule).
```
Comment (armchairlinguist) [inline docs/semgrep-ci/sample-ci-configs.md:257]:
```suggestion
5. Optional: Create a daily scheduled run for the custom pipeline on the main branch by [scheduling a pipleline in BitBucket](https://support.atlassian.com/bitbucket-cloud/docs/pipeline-triggers/#On-schedule).
```

This seems optional to me, so let's retain that note.

**Review by armchairlinguist — CHANGES_REQUESTED** [review]
Some of my changes are suggestions, but the SEMGREP_APP_TOKEN one is pretty important :)

**Comment by s-santillan** [comment]
With permission from @ekufta0530, I'll be taking this over the finish line :D

File: src/components/code_snippets/_bitbucket-semgrep-app-sast.mdx (hunk 1 above)
Comment (armchairlinguist) [reply src/components/code_snippets/_bitbucket-semgrep-app-sast.mdx:13]: Yeah, good point - we may need to not only restore this but make sure it's in all the right places.

Comment (ekufta0530) [reply src/components/code_snippets/_bitbucket-semgrep-app-sast.mdx:13]: @s-santillan SEMGREP_APP_TOKEN should still be in there. That is just my mistake.

Comment (s-santillan) [reply src/components/code_snippets/_bitbucket-semgrep-app-sast.mdx:13]: Added it into the 3 events. Resolving.

**Comment by s-santillan** [comment]
@armchairlinguist Vale doing its job :D

**Pull request merged**